### PR TITLE
NIFI-1745: Refactor how revisions are handled at NCM/Distributed to Node

### PR DESCRIPTION
**Pull request by markap14**
This is part 2 of NIFI-1745, which provides user info to revision claims/locks and removes the cluster context, as it is no longer necessary

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-optimistic-locking/src/main/java/org/apache/nifi/web/revision/NaiveRevisionManager.java [hunk 1]
```diff
@@ -423,7 +440,7 @@ public boolean requestWriteLock(final Revision proposedRevision) throws ExpiredR
                         throw ise;
                     }
 
-                    if (stamp.getClientId() == null || stamp.getClientId().equals(proposedRevision.getClientId())) {
+                    if (stamp.getUser() == null || stamp.getUser().equals(user)) {
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-optimistic-locking/src/main/java/org/apache/nifi/web/revision/NaiveRevisionManager.java]: I think we need to be checking both the user and the client id here. To protect against the case when the same user has multiple clients open (like multiple browser windows). One client shouldn't be able to assume the transaction from another client. Should also have a unit test for this scenario.

**Comment by mcgilman** [comment]
@markap14 When canceling revision claims for Connectable components in the verifyXxx() methods, DAO access will not return null so you do not need to check for it there. A ResourceNotFoundException will be thrown. Is that acceptable given it's usage?

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-optimistic-locking/src/main/java/org/apache/nifi/web/revision/NaiveRevisionManager.java (hunk 1 above)
Comment (markap14) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-optimistic-locking/src/main/java/org/apache/nifi/web/revision/NaiveRevisionManager.java]: @mcgilman good call. I have addressed the issue and updated the PR.

**Comment by markap14** [comment]
@mcgilman I agree, I do think it makes more sense to handle that in the Revision. Have updated the PR to back that out so that we can instead do it in the Resource in the next PR.

**Pull request closed without merging**